### PR TITLE
gh-47 add documentation to the docker images

### PR DESCRIPTION
**Pull request by lb324567**
tldr; 
I have tried to improve the readability of the documentation and also standardise the formatting.

Summary of changes and why:

Created READMEs for top levels and services:
- Hopefully improving readability
- Clarifies what instructions are for what area- kubernetes or docker
- Standardises the documentation across the different areas of the repo
- Adds additional detail to running the services, not all areas were documented with a README
- Explains what services are built using the files in that directory

Updated minimal READMEs
- Included more detailed explanations on some READMEs that had little content

Other:
- Spelling fixes
- Formatting standardisation
- Updated a compose file to keep port mapping consistent across all compose files for individual services

### CODE REVIEW

**Review by GCHQDeveloper314 — APPROVED** [review]

**Review by t92549 — DISMISSED** [review]
Way nicer now that everything is consistent

**Review by n3101 — DISMISSED** [review]

**Review by GCHQDeveloper314 — APPROVED** [review]

**Review by n3101 — DISMISSED** [review]

**Review by n3101 — DISMISSED** [review]

**Review by GCHQDeveloper314 — APPROVED** [review]

**Review by t92549 — APPROVED** [review]
